Add FeeSettings, LedgerHashes ledger object types

mDuo13 opened the pull request:
Adds two more types of ledger objects that weren't documented until now. Both of them are sort of internal tracking systems, but they are useful in their own ways.

This also corrects the LedgerEntryType and space key values in a couple places. Even though both types are enumerated using single ASCII characters in the code base, they get cast to UInt16's when serialized, so I represented them as `0x00__` values to make it clearer. (This explains the hashes of the singletons more accurately.)

Also, this diagram didn't make it into the doc, but it explains the discrepancy in the LedgerEntryType and Space Key values for the LedgerHashes and FeeSettings types:
![ledger-node-space-key-comparison](https://user-images.githubusercontent.com/7515597/32975867-afef2688-cbc0-11e7-82dc-1d51299ed41a.png)

mDuo13 committed:
Ledger objects: FeeSettings, LedgerHashes, related

mDuo13 committed:
Ledger format: Add ledger objects to type listing

mDuo13 committed:
ledger format: FeeSettings/LedgerHashes polish

JoelKatz commented on content/reference-ledger-format.md:
```diff
@@ -358,6 +362,118 @@ The ID of an Escrow object is the SHA-512Half of the following values put togeth
+
+| Name                | JSON Type | [Internal Type][] | Description            |
+|:--------------------|:----------|:------------------|:-----------------------|
+| `LedgerEntryType`   | String    | UInt16            | The value `0x0073`, mapped to the string `FeeSettings`, indicates that this object is a list of ledger hashes. |
```
The text here is not correct. Should read "indicates that this object contains the network fee settings" or something like that.

JoelKatz commented on content/reference-ledger-format.md:
```diff
@@ -358,6 +362,118 @@ The ID of an Escrow object is the SHA-512Half of the following values put togeth
+### FeeSettings ID Format
+
+The `FeeSettings` object ID is the hash of the `FeeSettings` space key (`0x0065`) only. This means that the ID of the Amendments object in a ledger is always:
+
```
"Amendments" should be "FeeSettings"

JoelKatz reviewed (approved):
:+1: (With correcting the two mistakes I noted.)

jbheron commented on content/reference-ledger-format.md:
```diff
@@ -358,6 +362,118 @@ The ID of an Escrow object is the SHA-512Half of the following values put togeth
+
+There are two formats for `LedgerHashes` object IDs, depending on whether the object is a "recent history" sub-type or a "previous history" sub-type.
+
+The **"recent history"** `LedgerHashes` object has an ID that is the SHA-512Half of the `LedgerHashes` space key (`0x0073`). In other words, the "recent history" always has the ID `B4979A36CDC7F3D3D5C31A4EAE2AC7D7209DDA877588B9AFC66799692AB0D66B`.
```
I hadn't heard of SHA-512Half, and had to look it up. Seeing it is a Ripple-specific term, I'd suggest adding a link to it.

jbheron reviewed (approved):
LGTM 

I added one suggestion about a link to SHA-512Half

mDuo13 committed:
Ledger format cleanup per reviews

- Fix typos in FeeSettings area pointed out by @joelkatz
- Add links to the explanation of SHA-512Half
- Consistent use of `CodeFont` for ledger object type names
